final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,181 @@
+    string = string[string.find('<') + 1:]
+    while string.find('<') - string.find('>') < 2:
+        string = string[string.find('>') + 2:]
+    string = string[string.find('>') + 1:string.find('<')]
+    return key[0].upper() + key[1:] + ':' + image + html.unescape(string)
+
+
+def set_start_setting():
+    # setup start settings
+    global args, limit
+    parser = argparse.ArgumentParser()
+    parser.add_argument("source", help="RSS URL")
+    parser.add_argument("--version", help="Print version info", action="store_true")
+    parser.add_argument("--json", help="Print result as JSON in stdout", action="store_true")
+    parser.add_argument("--verbose", help="Outputs verbose status messages", action="store_true")
+    parser.add_argument("--limit", help="Limit news topics if this parameter provided")
+    parser.add_argument("--date", help="Obtaining the cached news without the Internet")
+    args = parser.parse_args()
+    if args.limit:
+        limit = args.limit
+    if args.verbose:
+        logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+    else:
+        logging.basicConfig(filename="sample.log", level=logging.INFO)
+
+
+def beautiful_print_data(data, **kwargs):
+    logging.info(type(data))
+    if type(data) == feedparser.FeedParserDict:
+        if args.json:
+            data = to_json(data, **kwargs)
+            print(data)
+        else:
+            print()
+            for i in kwargs['feed']:
+                print(data['feed'][i])
+            print()
+            k = 1
+            for i in range(int(limit)):
+                for j in kwargs['entries']:
+                    print(text_for_print(j, data['entries'][i][j]))
+                print()
+            print()
+            print('Links:')
+            for i, v in enumerate(links):
+                print(f'[{i + 1}]{v}')
+            print()
+    elif type(data) == dict:
+        print()
+        for i in kwargs['feed']:
+            print(data[i])
+        print()
+        k = 1
+        for i in range(int(limit)):
+            for j in kwargs['entries']:
+                print(text_for_print(j, data['items'][i][j]))
+            print()
+        print()
+        print('Links:')
+        for i, v in enumerate(links):
+            print(f'[{i + 1}]{v}')
+        print()
+    elif type(data) == str:
+        print(data)
+
+
+def add_feed_to_file(json_data):
+    date_str = datetime.datetime.now().date().strftime('%Y%m%d')
+    try:
+        with open('cache.txt', 'r') as fin:
+            lines = fin.readlines()
+    except FileNotFoundError:
+        lines = []
+    with open('cache.txt', 'w') as file:
+        if not lines:
+            file.write(datetime.datetime.now().date().strftime('%Y%m%d') + ' ' + str(json_data) + '\n')
+        for line in lines:
+            if date_str in line:
+                file.write(datetime.datetime.now().date().strftime('%Y%m%d') + ' ' + str(json_data) + '\n')
+            else:
+                file.write(line)
```

Comment:
Instead of constantly reading file and rewriting it afterwards it is better to open this file using `append` mode.